source/extensions/filters/udp/udp_proxy/udp_proxy_filter.cc

```diff
@@ -171,6 +174,33 @@ UdpProxyFilter::ActiveSession::ActiveSession(ClusterInfo& cluster,
       .connections()
       .inc();
 
+  if (use_original_src_ip_) {
+    ENVOY_LOG(debug, "The original src is enabled for address {}.",
+              addresses_.peer_->asStringView());
+
+    const auto host_addr = host_->address()->ip();
+    const int trans_value = 1;
+    Api::SysCallIntResult result;
+
+    if (host_addr->version() == Network::Address::IpVersion::v6) {
+      // set ipv6 option
+      result = Network::SocketOptionImpl::setSocketOption(*socket_, ENVOY_SOCKET_IPV6_TRANSPARENT,
```

Comment:
Please see the other socket options and how we create platform/IP version agnostic factories for them. We should do this for the transparent option as well.

Reply:
Ok. I will check the other socket options and revise my patch.